lib/hammer_cli_foreman/task_helper.rb

```diff
@@ -0,0 +1,180 @@
+require 'fileutils'
+
+module HammerCLIForeman
+  module TaskHelper
+    class PluginDraft
+      attr_reader :name, :path
+
+      def initialize(name, path, options = {})
+        @plain_name = name
+        @name = "hammer_cli_foreman_#{name}"
+        @path = File.expand_path(@name.tr('_', '-'), path)
+        @capitalized_bits = name.split('_').map(&:capitalize)
+        @plugin_namespace = "HammerCLIForeman#{@capitalized_bits.join}"
+        @core_location = File.expand_path('../../', File.dirname(__FILE__))
+        @options = options
+      end
+
+      def build
+        FileUtils.mkpath("#{@path}/lib/#{@name}")
+        FileUtils.mkpath("#{@path}/config")
+        FileUtils.mkpath("#{@path}/test")
+        FileUtils.mkpath("#{@path}/locale")
+      end
+
+      def fill(&block)
+        FileUtils.cd(@path) do
+          self.instance_exec(&block)
+        end
+      end
+
+      private
+
+      def cp_license
+        FileUtils.cp(File.expand_path('LICENSE', @core_location), FileUtils.pwd)
+      end
+
+      def mk_config
+        File.open("config/foreman_#{@plain_name}.yml", 'w') do |config|
+          config.write <<-EOF
+:foreman_#{@plain_name}:
+  :enable_module: true
+EOF
+        end
+      end
+
+      def mk_readme
+        File.open('README.md', 'w') do |readme|
+          readme.write <<-EOF
+# TODO
+EOF
+        end
+      end
+
+      def mk_gemfile
+        File.open('Gemfile', 'w') do |gemfile|
+          gemfile.write <<-EOF
+source "https://rubygems.org"
+
+gemspec
+EOF
+        end
+      end
+
+      def mk_gemspec
+        File.open("#{@name}.gemspec", 'w') do |gemspec|
+          gemspec.write <<-EOF
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require '#{@name}/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = '#{@name}'
+  spec.version       = #{@plugin_namespace}.version.dup
```

Comment:
I think this should be:  `#{@plugin_namespace.version.dup}`

Reply:
Actually, if we do that, then `spec.version` will be hardcoded to the current one `0.0.1`.

But if we leave it in this way, it will be as we do in most plugins. So the new version is picked automatically when the gem is being built.